### DISCUSSION TITLE
Offline Mode?

### DISCUSSION THREAD

**Discussion by nathvi** [elm-time/elm-time#9] [Ideas]
Plans to support offline mode?

Basically, can edit and play around with a cached version, and then push changes to backend once connection is available again?

**Comment by Viir** [comment]
Yes, offline mode already works.
If you want to test without a network, just run a server on your machine.
So you can develop and test offline and then deploy to a remote system later.

To test changes in your app code, you have two ways with the current version:

+ Stop the `run-server` server command and rerun it.
+ Use the `deploy` command on the existing instance. This way, you can also test the whole deployment, including migrations.

The options on the `run-server` support replicating from a backup and deploying via options, so rerunning the same single command is enough to merge (via a deploy) your changes with the state that you have copied from a production system.
